bodhi/server/models.py

```diff
@@ -3653,6 +3653,41 @@ def comment_on_test_gating_status_change(target, value, old, initiator):
                 author="bodhi",
             )
 
+    def move_to_stable(self, db):
```

Comment:
This is misleading since it will not actually move the update to stable, but request the update to be move to stable by the next compose. 
The effect is that bodhi will tag the koji build with the release `Pending stable tag`. For the rawhide use case we still need to find out how the update status will be updated to stable after the nightly compose.

Maybe we should keep the name as `approve_testing` or something in that spirit.

Reply:
This makes me think that this is not working or going to work as we want.

If you check: ``koji -s https://koji.stg.fedoraproject.org/kojihub buildinfo python-arrow-0.12.1-10.fc31`` you will see that this build is still in ``f31-updates-testing`` while the build is 8 days old.

If the move to stable only happens at the compose time, since we are not doing compose in bodhi for rawhide, it means updates will be stuck in ``f31-updates-testing``, potentially, in a pending state in the UI, but nothing moves the update further :(

Reply:
yes also the f31 release in staging is not configure properly, the `Pending stable tag` should be `f31-pending` and not `f31-updates-testing` so that the builds get signed by robosignatory and tagged with the `f31` tag.

I guess we need a consumer that listen to the end of compose messages, and set the updates status to `STABLE` if the compose was successful.

Reply:
This means we need to figure out exactly which update went into every compose.
To avoid this work which is tedious and error-prone, we agreed that we would mark the updates as STABLE once they are available in the rawhide buildroot, just as it is today (once the build is in the buildroot, there is nothing else to consider for rawhide).

So the flow is:
- build in koji
- automatically create an update in bodhi in a PENDING state
- query greenwave to know if that update should be gated or not (did they opt-in?) if not, skip the next step
- wait for greenwave to announce its decision
- inform the user about greenwave's decision by a comment on the udpate
- if greenwave lets the update go through, push the update to the ``-pending`` tag and mark the update as STABLE.

Reply:
> This means we need to figure out exactly which update went into every compose.
> To avoid this work which is tedious and error-prone, we agreed that we would mark the updates as STABLE once they are available in the rawhide buildroot, just as it is today (once the build is in the buildroot, there is nothing else to consider for rawhide).
> 
> So the flow is:
> 
>     * build in koji
> 
>     * automatically create an update in bodhi in a PENDING state
> 
>     * query greenwave to know if that update should be gated or not (did they opt-in?) if not, skip the next step
> 
>     * wait for greenwave to announce its decision
> 
>     * inform the user about greenwave's decision by a comment on the udpate
> 
>     * if greenwave lets the update go through, push the update to the `-pending` tag and mark the update as STABLE.

Ha ok yes that makes things simpler :-). So we just need something like that in the approve_testing script : 

```
if update.release.mandatory_days_in_testing == 0:
    update.status =  UpdateStatus.stable

```
after this line https://github.com/fedora-infra/bodhi/blob/develop/bodhi/server/scripts/approve_testing.py#L105.

We have to keep in mind that the build might not be in stable tho depending on the signing status. Also I think users are used to be have the build available in the repo when it is marked as stable in bodhi so that might get them confused. 

Maybe we should approach the packagers that have originally commented on the Change proposal and get there inputs on these decisions.